### PR TITLE
Allow sorting of PVR Recordings and optionally hide watched items.

### PR DESCRIPTION
**Pull request by Selaron**
Implemented a new menu on PVRRecordings view:
Group Sorting:
 ( ) By name 
    -> Sorts the recordings by title (asc) and starttime (desc). Newest per title are thus visible on top
 ( ) By date added
    -> Sorts the recordings by starttime (desc) only, newest at the top.
 ( ) Unsorted
   -> Shows the recordings in same order as before as delivered by Kodi.

Additional Filter Checkbox
  [ ] Hide watched
  -> This is a heuristic that tries to hide recordings already watched:
      If recording.playcount is greater than 0, hide.
      If resume stamp is greater than tv show duration, hide.
      Tv show duration is typically smaller than total recording length as there are some minutes before and after the tv show recorded.

### CODE REVIEW

**Comment by Selaron** [comment]
This implements #385

File: app/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -325,6 +325,7 @@
     <string name="sort_by_length">Nach Länge</string>
     <string name="sort_by_date_added">Nach Hinzufügungsdatum</string>
     <string name="sort_by_last_played">Zuletzt gespielt</string>
+    <string name="unsorted">Nicht Sortiert</string>
```
Comment (Selaron) [inline app/src/main/res/values-de/strings.xml]: "Sortiert" should be lower case: "sortiert"

File: app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java [hunk 2]
```diff
@@ -134,10 +228,90 @@ public void onSuccess(List<PVRType.DetailsRecording> result) {
...
+
+                setupRecordingsGridview(finalResult);
                 swipeRefreshLayout.setRefreshing(false);
             }
 
+            private List<PVRType.DetailsRecording> filter(List<PVRType.DetailsRecording> itemList) {
+                SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getActivity());
+                boolean hideWatched = preferences.getBoolean(Settings.KEY_PREF_PVR_RECORDINGS_FILTER_HIDE_WATCHED, Settings.DEFAULT_PREF_PVR_RECORDINGS_FILTER_HIDE_WATCHED);
+
+                boolean filter = hideWatched;
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java]: Redundant var, can use `hideWatched`

File: app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java [hunk 3]
```diff
@@ -134,10 +228,90 @@ public void onSuccess(List<PVRType.DetailsRecording> result) {
...
+
+                List<PVRType.DetailsRecording> result;
+                if(filter) {
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java]: Please put a space between `id` and `(`

File: app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java [hunk 4]
```diff
@@ -134,10 +228,90 @@ public void onSuccess(List<PVRType.DetailsRecording> result) {
...
+                    result = new ArrayList<>(itemList.size());
+                    for(PVRType.DetailsRecording item:itemList) {
+                        if(hideWatched) {
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java]: Redundant check, can be removed

File: app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java (hunk 3 above)
Comment (SyncedSynapse) [reply app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java]: By the way, the code would be cleaner if at the very start of this function you did
`if (hideWatched) return itemList`

And them just include the code to filter the list.

File: app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java [hunk 5]
```diff
@@ -134,10 +228,90 @@ public void onSuccess(List<PVRType.DetailsRecording> result) {
...
+                        };
+                        Collections.sort(itemList, comparator);
+                        break;
+                    case Settings.SORT_BY_NAME:
+                        // sort by recording title and start time
+                        comparator = new Comparator<PVRType.DetailsRecording>() {
+                            @Override
+                            public int compare(PVRType.DetailsRecording a, PVRType.DetailsRecording b) {
+                                int result = a.title.compareToIgnoreCase(b.title);
+                                if(0 == result) { // note the yoda condition ;)
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java]: noted.
And please, a space between the if and the condition.

File: app/src/main/res/values-de/strings.xml (hunk 1 above)
Comment (SyncedSynapse) [reply app/src/main/res/values-de/strings.xml]: Can't you change it?

**Comment by SyncedSynapse** [comment]
To be honest, i don't have my PVR setup working, so i can't test this, but the code looks good, so i'm going to trust you and merge this.

Could you just check the (small) comments i made, and improve them.

File: app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java (hunk 2 above)
Comment (Selaron) [reply app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java]: I've had in mind here that there may be additional conditions added later on - e.g. an outstanding PR for #695 would add a `boolean hasSearchFilter` condition.

**Comment by SyncedSynapse** [comment]
Merging this, though there's still a redundant check...

**Pull request merged**